cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,238 @@
+        return view
+
+    def _execute_action_button(self, button, option, handle_alert=True, **kwargs):
+        target = kwargs.get("target", None)
+        provider = kwargs.get("provider", None)
+        desired_state = kwargs.get("desired_state", None)
+
+        view = self._execute_button(button, option, handle_alert=handle_alert)
+
+        if desired_state:
+            self._wait_for_state_change(desired_state, target, provider, view)
+        elif handle_alert:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+
+    @variable(alias='ui')
+    def power_state(self):
+        view = navigate_to(self, "Details")
+        return view.entities.power_management.get_text_of("Power State")
+
+    @variable(alias='ui')
+    def part_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Part Number")
+
+    @variable(alias='ui')
+    def serial_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Serial Number")
+
+    @variable(alias='ui')
+    def description(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Description")
+
+    @variable(alias='ui')
+    def manufacturer(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Manufacturer")
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
+        """Wait for PhysicalSwitch to come to desired state. This function waits just the needed amount of
+           time thanks to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            target (str): The name of the method that most be used to compare with the desired_state
+            view (object): The view that most be refreshed to verify if the value was changed
+            provider (object): 'LenovoProvider'
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+            delay (int): Specify amount of time (in seconds) to repeat each time.
+        """
+
+        def _is_state_changed():
+            self.refresh(provider, handle_alert=True)
+            return desired_state == getattr(self, target)()
+
+        wait_for(_is_state_changed, fail_func=view.browser.refresh, num_sec=timeout, delay=delay)
+
+    @property
+    def exists(self):
+        """Checks if the physical_switch exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def get_db_id(self):
```

Comment:
**REQUIRED**
This property should have name `db_id`. The attribute should be renamed to `_db_id`.

Reply:
Done.